### ISSUE TITLE
How to run this project

### ISSUE THREAD

**Issue by Antactz** [oci-labs/node-opendds#24]
Hello, Now I want to use a tool to realize the function of OpenDDS connecting with web pages, similar to Ros2-Web-Bridge. Is this project the same? Also, when I build this project, I ran the code `Node-gyp configure build`. After successful operation, I generated the build folder, there are many files below, including `node_opendds.node`, etc. But Then I don't know how to operate, how to run the contents of the test folder, please help me, thank you very much!

**Comment by mitza-oci** [comment]
Try the included `run_test.pl` script to run things in the `test` directory.
There is also a larger demo application in its own repository https://github.com/oci-labs/iiot-demo-opendds

**Comment by Antactz** [comment]
Thank you very much for your reply. I tried to run the `run_test.pl` file, but the error was reported as follows:

`ben@ubuntu:~/Desktop/node-opendds$ cd test/
ben@ubuntu:~/Desktop/node-opendds/test$ ./run_test.pl 
/home/ben/Desktop/node-opendds/test/test_publisher: error while loading shared libraries: libv8stubs.so.3.15.0: cannot open shared object file: No such file or directory
ERROR: test_publisher returned 127
(106694|106694) WARNING: Could not find FQDN. Using "ubuntu" as fully qualified hostname, please correct system configuration.
(106702|106702) WARNING: Could not find FQDN. Using "ubuntu" as fully qualified hostname, please correct system configuration.
ERROR: /usr/local/bin/node timedout
(106694|106711) we did not try to re-establish connection on transport: _OPENDDS_InfoRepoBITTCPTransportInst to 127.0.1.1:38381.
ERROR: /usr/local/bin/node returned -1
INFO: /home/ben/Downloads/OpenDDS/bin/DCPSInfoRepo being killed.
InfoRepo_Shutdown: shutting down on signal 15
(106694|106700) ERROR: DCPS_IR_Domain::remove_dead_participants () Removing dead participant 0xff8a84c0 id 01030000.259cabfc.00000002.000001c1(aef32e69)
(106694|106700) ERROR: WriteDataContainer::dispose, The instance(handle=2) is not registered yet.
(106694|106700) ERROR: DataWriterImpl::dispose: dispose failed.
(106694|106700) ERROR: DCPS_IR_Domain::dispose_participant_bit Unable to dispose for participant 01030000.259cabfc.00000002.000001c1(aef32e69) handle 2.
(106694|106694) WARNING: DataLink[0x55bdff8aa370]::~DataLink() - link still in use by 4 entities when deleted!
ben@ubuntu:~/Desktop/node-opendds/test$ `

I don't know what's wrong with that

**Comment by simpsont-oci** [comment]
I know this is an old issue, so hopefully you've figured out what you needed before now but I'll reply anyway. We've updated our README to give clearer instructions for getting the tests build and running. The error you were seeing was due to the fact that the v8 stubs shared library (used by the C++ test applications) was not found on the library path (either because it wasn't built or the path was incorrect). Hopefully if you follow the updated run instructions, all the environment variables and makefiles will allow you to successfully run the tests.
